### PR TITLE
Allow WCS equality checks to be configured

### PR DESCRIPTION
**Pull request by keflavich**
This will allow small errors in WCS precision to be ignored so that cubes on
the same grid to within some tolerance can be compared.

WIP

cc @astrofrog

### CODE REVIEW

File: spectral_cube/tests/test_masks.py [hunk 1]
```diff
@@ -256,6 +256,30 @@ def test_wcs_validity_check():
...
+    mask = BooleanArrayMask(data>0, cube._wcs)
+
+    # but if it's not exactly equal, an error should be raised at this step
+    with pytest.raises(ValueError) as exc:
+        cube = cube.with_mask(mask)
+    assert exc.value.args[0] == "WCS does not match mask WCS"
+
+    # this one should work though
+    cube = cube.with_mask(mask, wcs_tolerance=1e-4)
+    assert cube._wcs_tolerance == 1e-4
```
Comment (astrofrog) [inline spectral_cube/tests/test_masks.py]: If a user does:

```
cube = cube.with_mask(mask, wcs_tolerance=1e-4)
cube = cube.with_mask(mask, wcs_tolerance=1e-6)
```

The internal state will end up different from:

```python
cube = cube.with_mask(mask, wcs_tolerance=1e-6)
cube = cube.with_mask(mask, wcs_tolerance=1e-4)
```

What about requiring the user to set wcs_tolerance as a settable property on the cube before adding masks?

Comment (keflavich) [inline spectral_cube/tests/test_masks.py]: In the above examples, you have 3 different cube objects in each case.  I advocate instead:

```
masked_cube1 = cube.with_mask(mask, wcs_tolerance=1e-4)
masked_cube2 = cube.with_mask(mask, wcs_tolerance=1e-6)
```

However, in the case that you want to keep adding masks, you may have to increase the tolerance...
```
masked_cube1 = cube.with_mask(mask, wcs_tolerance=1e-6)
masked_cube2 = masked_cube1.with_mask(mask2, wcs_tolerance=1e-5)
```

Does this make sense?

I'm trying to treat `cube` objects as immutable as much as possible

Comment (astrofrog) [inline spectral_cube/tests/test_masks.py]: Ah sorry, I completely forgot we are creating a new cube every time

**Comment by astrofrog** [comment]
This looks good to me

**Comment by keflavich** [comment]
@e-koch would be helpful to get an independent review from you on the docs

File: docs/masking.rst [hunk 2]
```diff
@@ -153,3 +154,49 @@ Optionally, a redundant Stokes axis can be added to match the original CASA imag
...
+    PC2_1 PC2_2 PC2_3  : 0.0  1.0  0.0  
+    PC3_1 PC3_2 PC3_3  : 0.0  0.0  1.0  
+    CDELT : -1.3888888888889999e-05  1.3888888888889999e-05  300.00000001056611  
+    NAXIS    : 0 0
+
+In order to compose masks from these, we need to set the ``wcs_tolerance`` parameter::
+
+    >>> masked_cube13co = cube13co.with_mask(mask_12co, wcs_tolerance=1e-3)  # doctest: +SKIP
+
+which in this case will check fro equality at the 1e-3 level, which truncates
```
Comment (e-koch) [inline docs/masking.rst]: fro -> for

**Comment by e-koch** [comment]
:+1:  Only found the one typo.

**Pull request merged**